test/run_test.py

```diff
@@ -532,10 +532,13 @@ def get_slow_tests_based_on_S3() -> List[str]:
 
 
 def get_test_case_args(test_module, using_pytest) -> List[str]:
-    if test_module not in SPECIFIED_TEST_CASES_DICT:
-        sys.exit(f'Warning! Test module {test_module} is not found in the specified tests dict. This should never'
-                 'happen as we make a check for that before entering this function.')
     args = []
+    if test_module not in SPECIFIED_TEST_CASES_DICT:
```

Comment:
maybe clarify by adding a comment that this should be triggered in the bring_to_front case/if you don't want to make the distinction, I think this if statement could be merged with the second